### PR TITLE
Provide `botocore` extra

### LINKED ISSUE #6: Provide option to tie installable version of `botocore` to that of `botocore-stubs`
This request is inspired by the optional `boto3` extra offered by `boto3-stubs`. Selecting the extra sets a hard pin on the versions of `boto3` and `botocore`, effectively providing convenient synchronisation of those packages' installed versions:
`pip install boto3==1.28.60 boto3-stubs[boto3]` will install version 1.28.60 of both boto3 and boto3-stubs. Despite there being available a more recent version of boto3-stubs.

Such a simple mechanism might also be offered by `botocore-stubs`: A new extra named `botocore` would set a hard pin to the respective version of `botocore`.

### PR DESCRIPTION
**Pull request by jakob-keller**
Closes #6

### CODE REVIEW

**Comment by jakob-keller** [comment]
I'm not sure if I need to update the poetry lock file. It looks a bit dated to me.

File: poetry.lock [hunk 1]
```diff
@@ -1,10 +1,9 @@
-# This file is automatically @generated by Poetry 1.4.2 and should not be changed by hand.
+# This file is automatically @generated by Poetry 1.6.1 and should not be changed by hand.
```
Comment (jakob-keller) [inline poetry.lock:1]: I had to update the lock file to add the optional dependency.

**Comment by jakob-keller** [comment]
@vemel: If this PR becomes stale, just let me know and I'll do a rebase.

**Comment by vemel** [comment]
Thank you a lot for your contribution! Sorry it took so long to merge it.

**Comment by jakob-keller** [comment]
> Thank you a lot for your contribution! Sorry, it took so long to merge it.

Always happy to contribute!

**Pull request merged**